cli/2024-04-azure-workload-identity.md

```diff
@@ -0,0 +1,233 @@
+
+**Sample Input:**
+
+See above
+
+**Sample Output:**
+
+See above
+
+**Sample Recipe Contract:**
+
+n/a
+
+## Design
+
+### High Level Design
+
+During installation, if the user enables the Azure Workload Identity feature, the Radius control plane (bicep-de, ucp, applications-rp) pods will be annotated with the `azure.workload.identity/use: "true"` label. The AzWI mutating admission webhook will see this label on the pods and will project a signed service account token to each of the pods. The signed service account token and the credential registered by the user will be used by Radius to authenticate with Azure.
+
+Please see https://azure.github.io/azure-workload-identity/docs/concepts.html for more information on how the AzWI mutating admission webhook works.
+
+### Architecture Diagram
+
+![Architecture Diagram](./2024-04-azure-workload-identity/architecture.png)
+
+### Detailed Design
+
+#### Advantages (of each option considered)
+
+n/a
+
+#### Disadvantages (of each option considered)
+
+n/a
+
+#### Proposed Option
+
+n/a
+
+### API design
+
+We should not need to make any functional changes to the Radius API design for this feature as long as we re-use the `ClientID` field for Azure credentials. We should update the documentation for the API and maybe some of the field names to represent that the `ClientID` field can be used for both service principal and Azure Workload Identity. We should also ensure that the `ClientSecret` field is not marked as required anywhere in the API spec since it is not required for Azure Workload Identity.
+
+
+### CLI Design
+
+`rad init --full` will need to be updated to ask the user to choose between service principal auth and Azure Workload Identity.
+
+We will need to update the `rad credential register azure` command to not require a client secret when registering the Entra ID Application Client ID.
+
+We will need to update the Radius Helm chart to allow the user to enable Azure Workload Identity during installation with the `global.azureWorkloadIdentity.enabled` value. We will need to update the installation flow (for `rad init` and `rad install kubernetes`) to annotate the Radius control plane pods with the `azure.workload.identity/use: "true"` label if the user sets this value in the Helm chart.
+
+### Implementation Details
+
+#### UCP
+
+UCP should be updated to read the Entra ID Application Client ID from the credential store. It should then authenticate with Azure using this Client ID and the signed service account token.
+
+The credential storage and management code will need to be updated to store a ClientID without a ClientSecret.
+
+#### Bicep
+
+n/a
+
+#### Deployment Engine
+
+Deployment Engine should be updated to read the Entra ID Application Client ID from the credential store. It should then authenticate with Azure using this Client ID and the signed service account token.
+
+#### Core RP
```

Comment:
We should also check how Terraform recipe deployment gets impacted by this since these credentials are used for Azure deployment via Terraform as well.

Reply:
https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/aks_workload_identity

Reply:
I will add some info about terraform changes here